lib/Doctrine/Record/Filter/Compound.php

```diff
@@ -56,6 +56,10 @@ public function init()
     public function filterSet(Doctrine_Record $record, $name, $value)
     {
         foreach ($this->_aliases as $alias) {
+            // The relationship must be fetched in order to check the field existance.
+            // Related to PHP-7.0 compatibility so an explicit call to method get is required.
+            $record[$alias];
```

Comment:
Not sure what this is doing, can you clarify this part? @alquerci ?

Reply:
@thePanz see the deep explanation on https://github.com/FriendsOfSymfony1/symfony1/pull/184#issuecomment-406818771

Reply:
ok, would it be possible to annotate the `protected $_aliases` with the right type-hint (line 35)?

We should also update the `@param mixed $name` as I guess it should be `string|int|null`? (not sure if php allows using objects as array keys).

wdyt?

Reply:
1. Yes, as `string[]`.
2. This comment is probably redundant.
     1. Let's remove it, then ask ourselves what information is missing?
     2. Does adding `@param string $name`, add information not said by the code itself?

Reply:
Adding the right annotation to the `@param string|int|null $name` would allow phpstan to find issues in the caller.
`mixed` is somehow not correct according to https://php.watch/versions/8.0/mixed-type

Reply:
https://github.com/FriendsOfSymfony1/doctrine1/pull/56/commits/3c1a6cf1a2ef1d87730d87361db5f7b54d3ab0fd

wdyt?

---

Related to that, I also made fixing of the code structure about record filter, on a dedicated PR as this one is enough big.
https://github.com/alquerci/doctrine1/pull/1

Reply:
@alquerci I would still add the phpdoc to the $name parameter